reflex/app.py

```diff
@@ -436,8 +444,17 @@ def add_custom_404_page(
             meta=meta,
         )
 
-        froute = format.format_route
-        self.pages[froute(constants.SLUG_404)] = component
+        # format 404 route
+        route = format.format_route(constants.SLUG_404)
+
+        # add the page
+        self.pages[route] = component
+
+        # Add the load events.
+        if on_load:
```

Comment:
We should refactor this with the `add_page` method so we're not duplicating code. Is there any reason this method can't just call `add_page`?